src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -706,8 +721,15 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	}
 
 	public changeKernel(displayName: string): void {
-		this._contextsLoadingEmitter.fire();
-		this.doChangeKernel(displayName, true).catch(e => this.logService.error(e));
+		this._currentKernel = displayName;
+		this._currentKernelAlias = this.context?.serverCapabilities.notebookKernelAlias;
+		if (this.kernelAliases.includes(this.currentKernelAlias) && displayName === this.currentKernelAlias) {
+			this.doChangeKernel(displayName, true).catch(e => this.logService.error(e));
+		} else {
+			this._currentKernelAlias = undefined;
+			this._contextsLoadingEmitter.fire();
+			this.doChangeKernel(displayName, true).catch(e => this.logService.error(e));
+		}
```

Comment:
Why would kernelAliases not include this.currentKernelAlias?

Reply:
Yes, the only time it would not include is if it was undefined, I believe the second half of the condition is all I need.